src/Dependencies/Collections/SegmentedList`1.cs

```diff
@@ -855,6 +873,36 @@ public int LastIndexOf(T item, int index, int count)
             return SegmentedArray.LastIndexOf(_items, item, index, count);
         }
 
+        public int LastIndexOf(T item, int index, int count, IEqualityComparer<T>? comparer)
+        {
+            if ((Count != 0) && (index < 0))
+            {
+                ThrowHelper.ThrowIndexArgumentOutOfRange_NeedNonNegNumException();
+            }
+
+            if ((Count != 0) && (count < 0))
+            {
+                ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.count, ExceptionResource.ArgumentOutOfRange_NeedNonNegNum);
+            }
+
+            if (_size == 0)
```

Comment:
Are _size and Count the same? If so, maybe move this up two conditions and eliminate the first half of the conditionals?

Reply:
➡️ The order here is sensitive with respect to the specific exceptions thrown (as part of matching the behavior of `List<T>`).

Reply:
Yes, but if _size and Count are the same, there would be no case in which it would matter if this case is here or at the top.

Reply:
➡️ Makes sense; now updated.